integration/integration_test.go

```diff
@@ -7223,6 +7226,165 @@ func testSFTP(t *testing.T, suite *integrationTestSuite) {
 	require.Equal(t, testFilePath, sftpEvent.GetString(events.SFTPPath))
 }
 
+func testAgentlessConnection(t *testing.T, suite *integrationTestSuite) {
+	// create Teleport instance
+	teleInst := suite.newTeleport(t, nil, true)
+	t.Cleanup(func() {
+		require.NoError(t, teleInst.StopAll())
+	})
+
+	tc, err := teleInst.NewClient(helpers.ClientConfig{
+		Login:   suite.Me.Username,
+		Cluster: helpers.Site,
+		Host:    Host,
+	})
+	require.NoError(t, err)
+
+	// get OpenSSH CA public key and create host certs
+	ctx := context.Background()
+	authClient := teleInst.Process.GetAuthServer()
+	openSSHCA, err := authClient.GetCertAuthority(ctx, types.CertAuthID{
+		Type:       types.OpenSSHCA,
+		DomainName: helpers.Site,
+	}, false)
+	require.NoError(t, err)
+
+	caCheckers, err := sshutils.GetCheckers(openSSHCA)
+	require.NoError(t, err)
+
+	key, err := native.GeneratePrivateKey()
```

Comment:
Don't we have a pool of keys we can pull from? I thought @jakule recently made some updates that reduced CPU usage and flakiness by dramatically reducing the amount of key generation we do.

Reply:
He told me recently to do it this way, as the `native` package will precompute keys when tests are being run IIRC